backend/src/test/java/com/glebabramov/backend/service/ResumeServiceTest.java

```diff
@@ -203,6 +206,118 @@ void createResume_shouldReturn201Created_andTheCreatedResume_ifUserIsAdmin() {
+			resumeRepository.save(resumeWithNonExistentUser);
+			MongoUser userWithNonExistentResume = new MongoUser(BASIC_USER_ID, "username", "password", "user", "non-existent-resume-id");
+			mongoUserRepository.save(userWithNonExistentResume);
+			ResumeRequest resumeUpdateRequest = new ResumeRequest(BASIC_USERS_ASSOCIATED_RESUME_ID, "Company name", Collections.emptySet());
+			//WHEN
+			ResponseStatusException expected = associatedResumeDoesNotExistException("non-existent-resume-id");
+			ResponseStatusException actual = assertThrows(ResponseStatusException.class, () -> resumeService.updateResume(resumeUpdateRequest, mockedPrincipal));
+			//THEN
+			assertEquals(expected.getClass(), actual.getClass());
+			assertEquals(expected.getMessage(), actual.getMessage());
+		}
+
+		@Test
+		@DirtiesContext
+		@DisplayName("...should update the resume, handle respective user reassignments and return the updated resume if the user is an admin and all operations have been verified")
+		void updateResume_shouldUpdateResumeHandleRespectiveUserReassignments_andReturnUpdatedResume_ifUserIsAdminAndAllOperationsHaveBeenVerified() {
+			//GIVEN
+			resumeRepository.save(standardResume);
+			resumeRepository.save(basicUsersResume);
+			mongoUserRepository.save(basicUser);
+			mongoUserRepository.save(adminUser);
+			resumeRepository.save(oldResume);
+			ResumeRequest resumeUpdateRequest = new ResumeRequest(TEST_RESUMES_ID, "Company name", Set.of(ADMIN_USER_ID));
+			//WHEN
+			Resume expected = new Resume(TEST_RESUMES_ID, "Company name", Set.of(ADMIN_USER_ID), false, false);
+			Resume actual = resumeService.updateResume(resumeUpdateRequest, mockedPrincipal);
+			MongoUser expectedSideEffect1 = new MongoUser(BASIC_USER_ID, basicUser.username(), basicUser.password(), basicUser.role(), STANDARD_RESUME_ID);
+			MongoUser actualSideEffect1 = mongoUserRepository.findById(BASIC_USER_ID).get();
+			MongoUser expectedSideEffect2 = new MongoUser(ADMIN_USER_ID, adminUser.username(), adminUser.password(), adminUser.role(), TEST_RESUMES_ID);
+			MongoUser actualSideEffect2 = mongoUserRepository.findById(ADMIN_USER_ID).get();
+			Resume expectedSideEffect4 = new Resume(BASIC_USERS_ASSOCIATED_RESUME_ID, oldResume.name(), Collections.emptySet(), oldResume.invitationSent(), oldResume.isPublished());
+			Resume actualSideEffect4 = resumeRepository.findById(BASIC_USERS_ASSOCIATED_RESUME_ID).get();
+			Resume expectedSideEffect3 = new Resume(STANDARD_RESUME_ID, standardResume.name(), Set.of(BASIC_USER_ID), standardResume.invitationSent(), standardResume.isPublished());
+			Resume actualSideEffect3 = resumeRepository.findById(STANDARD_RESUME_ID).get();
+			//THEN
+			assertEquals(expected, actual);
+			assertEquals(expectedSideEffect1, actualSideEffect1);
+			assertEquals(expectedSideEffect2, actualSideEffect2);
+			assertEquals(expectedSideEffect4, actualSideEffect4);
+			assertEquals(expectedSideEffect3, actualSideEffect3);
```

Comment:
I adressed the issue with the repository priviously, I guess you will adress it in your upcoming PR regarding test optimization.

Reply:
Yes, you addressed and I have rewritten most of the tests BUT I have decided to mock the complete behaviour of the repository saving something -> means it may return it after the save on respective finById, findByUsername a. s.o. but that way I may control that all the necessary invocations of side effects have been actually called. (see my SetUp function)